tests/test_BlkDiagMatrix.py

```diff
@@ -0,0 +1,910 @@
+import numpy as np
+from unittest import TestCase
+
+from aspire.utils.blk_diag_matrix import BlkDiagMatrix
+
+class BlkDiagMatrixTestCase(TestCase):
+    def setUp(self):
+        self.blk_a = BlkDiagMatrix.from_blk_diag([
+            np.array([[-0.30656809, -0.34287864, -0.00854488,  0.5275285 ],
```

Comment:
Do these have any specific properties or can they be randomly generated (with the same seed, of course)?

Reply:
I am working on this today, using random gaussian blocks from a fixed RandomState.  The basic arithmetic went quickly.  I have to do apply and solve yet.

Reply:
#156